src/services/keycloakService.ts

```diff
@@ -0,0 +1,122 @@
+import * as WebConfigAPI from '../services/WebConfigService';
+declare global {
+  interface Window {
+    Keycloak?: any;
+  }
+}
+/**
+ * Enum with possible authentication roles in keycloak
+ *
+ */
+export enum AUTH_ROLE {
+  Admin = 'admin',
+  User = 'user',
+  System = 'system-user',
+  Power = 'power-user',
+}
+
+/**
+ * Class managing authentication functionality
+ */
+class KeycloakService {
+  //We can't get KeycloakInstance type because of dynamic loading of keycloak library
+  private keycloakAuth: any;
+
+  private isKeycloakInitialized;
+
+  constructor() {
+    this.isKeycloakInitialized = this.init();
+  }
+
+  /**
+   * Initialize keycloak and create instance
+   *
+   * @returns Promise
+   */
+  private init(): Promise<any> {
+    const keycloakConfig = WebConfigAPI.getWebConfig().keycloak;
+
+    this.keycloakAuth = window.Keycloak({
+      url: keycloakConfig.url,
+      realm: keycloakConfig.realm,
+      clientId: keycloakConfig.clientId,
+    });
+
+    return new Promise((resolve, reject) => {
+      this.keycloakAuth
+        .init({ onLoad: 'check-sso', checkLoginIframeInterval: 1 })
```

Comment:
Default value for `checkLoginIframeInterval` is `5` ( https://github.com/keycloak/keycloak-documentation/blob/main/securing_apps/topics/oidc/javascript-adapter.adoc ), is there some reason to change it? If not, I would use the default value.